crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -1041,6 +1064,16 @@ fn unwrap_struct_type(typ: &HirType) -> Vec<(String, HirType)> {
     }
 }
 
+fn unwrap_array_element_type(typ: &HirType) -> HirType {
+    match typ.clone() {
+        HirType::Array(_, elem) => *elem,
+        HirType::Slice(elem) => *elem,
+        other => {
+            unreachable!("unwrap_array_element_type: expected an array or slice, found {:?}", other)
+        }
```

Comment:
This function needs to follow type variables as well, see e.g. `unwrap_struct_type`

Reply:
ah, done